llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -852,3 +871,13 @@ def _save_checkpoint(
                 if use_temp_dir:
                     shutil.rmtree(temp_save_dir)
         dist.barrier()
+
+        if dist.get_global_rank() == 0:
+            self.save_additional_contents(
```

Comment:
why does this need all these args? shouldn't it just need `state` and `temp_save_dir`? Also shouldn't this happen before upload and register?